packages/expo-router/src/__tests__/push.test.ios.tsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import { Text } from 'react-native';
+
+import { store } from '../global-state/router-store';
+import { router } from '../imperative-api';
+import Stack from '../layouts/Stack';
+import Tabs from '../layouts/Tabs';
+import { act, renderRouter, screen } from '../testing-library';
+import { Slot } from '../views/Navigator';
+
+it('stacks should always push a new route', () => {
```

Comment:
This test is a simplified version of #26669. There were two problems

1) It targeted based upon name, so it always targeted `(group)/_layout`
2) Once that was fixed it didn't target correctly based upon different params, e.g `user/2` was targeting the `<Stack />` inside `user/1/_layout` instead of `(group)/_layout`

Because of the incorrect target, Screens were being nested instead of pushed (hence why they switched instead of animated).

This test shows they are always being pushed